pkgs/os-specific/darwin/impure-cmds/default.nix

```diff
@@ -0,0 +1,34 @@
+{ lib, runCommandLocal }:
+
+# On darwin, there are some commands neither opensource nor able to build in nixpkgs.
+# We have no choice but to use those system-shipped impure ones.
+
+let
+  commands = {
+    ditto = "/usr/bin/ditto"; # ditto is not opensource
+    sudo  = "/usr/bin/sudo";  # sudo must be owned by uid 0 and have the setuid bit set
+  };
+
+  mkImpureDrv = name: path:
+    runCommandLocal "${name}-impure-darwin" {
+      __impureHostDeps = [ path ];
+
+      meta = {
+        platforms = lib.platforms.darwin;
```

Comment:
```suggestion
        platforms = with lib.platforms; [ darwin ];
```

Reply:
Inside `pkgs/os-specific/darwin`, all packages are using `platforms = lib.platforms.darwin;`, because obviously we will rarely add support to other platforms. Would it be better to follow the tradition？

Reply:
I agree with you. This is kinda bikesheeding.

Reply:
Actually, the original form does produce a valid platform list when introspected. I did not realize that.